### PR TITLE
pr93x-CaloSaturations-at-P5

### PR DESCRIPTION
**Pull request by rekovic**
pr93x CaloSaturations-at-P5

Small fixes to jets & sums saturation. 
- Now HT, MHT saturate in the same way as MET corresponding to fw
- Don't calibrate saturated jets for HT.
- Now use 0 instead of 111 as default MHT phi value when the MHT == 0.

### CODE REVIEW

**Comment by rekovic** [comment]
please test

File: L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2DemuxSumsAlgoFirmwareImp1.cc [hunk 1]
```diff
@@ -34,7 +34,7 @@ void l1t::Stage2Layer2DemuxSumsAlgoFirmwareImp1::processEvent(const std::vector<
   uint32_t mbp0(0), mbm0(0), mbp1(0), mbm1(0);
```
Comment (rekovic) [inline L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2DemuxSumsAlgoFirmwareImp1.cc]: Change `uint` and `uint32_t` to `unsigned int`

File: L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2EtSumAlgorithmFirmwareImp1.cc [hunk 2]
```diff
@@ -49,7 +49,7 @@ void l1t::Stage2Layer2EtSumAlgorithmFirmwareImp1::processEvent(const std::vector
     int32_t etem(0);
```
Comment (rekovic) [inline L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2EtSumAlgorithmFirmwareImp1.cc]: Same as above.... also int32_t change to int.

File: L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2JetSumAlgorithmFirmwareImp1.cc [hunk 3]
```diff
@@ -36,6 +36,8 @@ void l1t::Stage2Layer2JetSumAlgorithmFirmwareImp1::processEvent(const std::vecto
 
     int32_t hx(0), hy(0), ht(0);
     int32_t hxHF(0), hyHF(0), htHF(0);
+
+    bool satMht(0), satMhtHF(0), satHt(0), satHtHF(0);
   
     // loop over rings    
     for (unsigned absieta=1; absieta<=(uint)CaloTools::mpEta(CaloTools::kHFEnd); absieta++) {
```
Comment (rekovic) [inline L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2JetSumAlgorithmFirmwareImp1.cc]: Use `unsigned int`

File: L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2DemuxSumsAlgoFirmwareImp1.cc [hunk 4]
```diff
@@ -34,7 +34,7 @@ void l1t::Stage2Layer2DemuxSumsAlgoFirmwareImp1::processEvent(const std::vector<
...
   uint32_t ntow(0);
 
-  bool metSat(0);
+  bool metSat(0), metHFSat(0), mhtSat(0), mhtHFSat(0);
 
   // Add up the x, y and scalar components
   for (std::vector<l1t::EtSum>::const_iterator eSum = inputSums.begin() ; eSum != inputSums.end() ; ++eSum )
```
Comment (rekovic) [inline L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2DemuxSumsAlgoFirmwareImp1.cc]: Use range-based for loop.

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
+1

**Comment by rekovic** [comment]
This PR addresses the differences in Calo emulator vs Calo firmware.
If not much problem for integration, it would be good to have it in MC production.

File: L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2DemuxSumsAlgoFirmwareImp1.cc (hunk 1 above)
Comment (davidlange6) [inline L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2DemuxSumsAlgoFirmwareImp1.cc]: hi @rekovic - just to clarify - uint32_t is fine.

**Comment by davidlange6** [comment]
+1

**Pull request merged**